easybuild/easyblocks/m/molpro.py

```diff
@@ -0,0 +1,259 @@
+            launcher = self.toolchain.mpi_cmd_for('%x', self.cfg['parallel'])
+            launcher = launcher.replace(' %s' % self.cfg['parallel'], ' %n')
+
+            # patch CONFIG file to change LAUNCHER definition, in order to avoid having to start mpd
+            apply_regex_substitutions(cfgfile, [(r"^(LAUNCHER\s*=\s*).*$", r"\1 %s" % launcher)])
+
+            # reread CONFIG and log contents
+            cfgtxt = read_file(cfgfile)
+            self.log.info("Contents of CONFIG file:\n%s", cfgtxt)
+
+    def build_step(self):
+        """Custom build procedure for Molpro, unless it is a binary install."""
+        if not self.cfg['precompiled_binaries']:
+            super(EB_Molpro, self).build_step()
+
+    def test_step(self):
+        """
+        Custom test procedure for Molpro.
+        Run 'make quicktest, make test', but only for source install and if license is available.
+        """
+
+        # Only bother to check if the licence token is available
+        if os.path.isfile(self.license_token) and not self.cfg['precompiled_binaries']:
+
+            # check 'main routes' only
+            run_shell_cmd("make quicktest")
+
+            if build_option('mpi_tests'):
+                # extensive test
+                run_shell_cmd("make MOLPRO_OPTIONS='-n%s' test" % self.cfg['parallel'])
+            else:
+                self.log.info("Skipping extensive testing of Molpro since MPI testing is disabled")
+
+    def install_step(self):
+        """
+        Custom install procedure for Molpro.
+        For source install:
+        * put license token in place in $installdir/.token
+        * run 'make tuning'
+        * install with 'make install'
+        For binary install:
+        * run interactive installer
+        """
+
+        if self.cfg['precompiled_binaries']:
+            """Build by running the command with the inputfiles"""
+
+            change_dir(self.cfg['start_dir'])
+
+            for src in self.src:
+                if LooseVersion(self.version) >= LooseVersion('2015'):
+                    # install dir must be non-existent
+                    shutil.rmtree(self.installdir)
+                    cmd = "./{0} -batch -prefix {1}".format(src['name'], self.installdir)
+                else:
+                    cmd = "./{0} -batch -instbin {1}/bin -instlib {1}/lib".format(src['name'], self.installdir)
+
+                bindir = os.path.join(self.installdir, 'bin')
+                libdir = os.path.join(self.installdir, 'lib')
+
+                qa = [
+                    (r"Please give your username for accessing molpro\n", ''),
+                    (r"Please give your password for accessing molpro\n", ''),
+                    (r"Enter installation directory for executable files \[.*\]\n", bindir),
+                    (r"Enter installation directory for library files \[.*\]\n", libdir),
+                    (r"directory .* does not exist, try to create [Y]/n\n", ''),
```

Comment:
It looks like the 2021 version only asks the first two questions:
```
== 2024-05-22 15:34:18,323 run.py:501 INFO Output of './molpro-mpp-2021.2.0.linux_x86_64_sockets.sh ...' shell command (stdout + stderr):

Installation of Molpro files complete

Please give your username for accessing molpro

Please give your password for accessing molpro
Token download unsuccessful - please install manually
```
and the 2024 versions don't ask anything anymore:
```
== 2024-05-22 15:36:54,840 run.py:501 INFO Output of './molpro-mpp-2024.1.0.linux_x86_64_sockets.sh ...' shell command (stdout + stderr):

Installation of Molpro files complete
```
But I guess it doesn't do any harm by leaving this in.

Reply:
I prefer leaving what we had in place, especially if it doesn't cause harm...